connector_office_365_hr_leave/models/hr_leave.py

```diff
@@ -50,3 +51,39 @@ def _validate_leave_request(self):
             if not leave.meeting_id and leave.meeting_id.user_id:
                 continue
             leave._office_365_push()
+
+    @api.multi
+    def action_refuse(self):
+        # not sure about the copy but _remove_resource_leave searches through
+        # all leaves might be an issue with lots of data
+        current_employee = self.env['hr.employee'].search(
+            [('user_id', '=', self.env.uid)],
+            limit=1
+        )
+        for holiday in self:
+            if holiday.state not in ['confirm', 'validate', 'validate1']:
+                raise UserError(_(
+                    'Leave request must be confirmed or validated '
+                    'in order to refuse it.'))
+
+            if holiday.state == 'validate1':
+                holiday.write({
+                    'state': 'refuse',
+                    'first_approver_id': current_employee.id
+                })
+            else:
+                holiday.write({
+                    'state': 'refuse',
+                    'second_approver_id': current_employee.id
+                })
+            # Delete the meeting
+            if holiday.meeting_id:
+                holiday.meeting_id.with_context(
+                    user=holiday.meeting_id.user_id
```

Comment:
can we inject in context 'user' and call super().action_refuse()?

Reply:
@i-vyshnevska no sadly not. if you refuse multiple leaves which  belong to different people the user must be the owner of the calendar entry. 
loop before calling super which lead to the comment above.